src/ManifoldProjections.jl

```diff
@@ -20,8 +20,20 @@ abstract type Manifold
 end
 
 # fallback for out-of-place ops
+"Returns a point that corresponds to the retraction of the given point `x` back onto the Manifold `M`"
 retract(M::Manifold, x) = retract!(M, copy(x))
+"Retracts a given point `x` back onto the Manifold `M`"
+function retract!(M::Manifold, x) end
```

Comment:
IMO this type of comment is just noise: it doesn't tell you anything that you wouldn't get by reading the function name. Adding more comments is great when a code is finalized and won't change for ten years. I doubt this is the case with this library, so any comment means giving a false sense of a mature library, and means having to maintain it when we change things around.